### PR TITLE
make loose datetime regex more strict

### PR DESCRIPTION
**Pull request by snopoke**
http://manage.dimagi.com/default.asp?167670

buddy @sravfeyn 

@dannyroberts FYI

### CODE REVIEW

**Comment by kaapstorm** [comment]
:+1:

File: dimagi/ext/jsonobject.py [hunk 1]
```diff
@@ -77,7 +77,7 @@ def _unwrap(self, value):
 # i.e. I just removed (...)? surrounding the second two lines
 re_loose_datetime = re.compile(
     r'^(\d{4})\D?(0[1-9]|1[0-2])\D?([12]\d|0[1-9]|3[01])'
-    r'\D?([01]\d|2[0-3])\D?([0-5]\d)\D?([0-5]\d)?\D?(\d{3,6})?'
+    r'[ T]([01]\d|2[0-3])\D?([0-5]\d)\D?([0-5]\d)?\D?(\d{3,6})?'
```
Comment (kaapstorm) [inline dimagi/ext/jsonobject.py]: Do you think we should replace every `\D?` with ":" or "-"?

Comment (snopoke) [inline dimagi/ext/jsonobject.py]: I don't think so. Some really are optional like between the date and the time: https://pypi.python.org/pypi/iso8601

**Comment by dannyroberts** [comment]
Wow, I didn't realize the regex was quite that loose. I thought I was doing the logical equivalent by using the same regex, but I now realize where I went wrong is that jsonobject matches, tries to convert, and if it throws any exception silently moves on. So that regex was catching a lot of things that might not match. Anyhow, I think this is a good solution for now

**Pull request merged**